src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -340,35 +334,35 @@ private static uint HandleEventShutdownComplete(State state, ref ConnectionEvent
                 bidirectionalTcs.SetException(ExceptionDispatchInfo.SetCurrentStackTrace(new QuicOperationAbortedException()));
             }
 
-            return MsQuicStatusCodes.Success;
+            return QUIC_STATUS_SUCCESS;
         }
 
-        private static uint HandleEventNewStream(State state, ref ConnectionEvent connectionEvent)
+        private static unsafe int HandleEventNewStream(State state, ref QUIC_CONNECTION_EVENT connectionEvent)
         {
-            var streamHandle = new SafeMsQuicStreamHandle(connectionEvent.Data.PeerStreamStarted.Stream);
-            if (!state.TryQueueNewStream(streamHandle, connectionEvent.Data.PeerStreamStarted.Flags))
+            var streamHandle = new SafeMsQuicStreamHandle(connectionEvent.PEER_STREAM_STARTED.Stream);
+            if (!state.TryQueueNewStream(streamHandle, connectionEvent.PEER_STREAM_STARTED.Flags))
             {
                 // This will call StreamCloseDelegate and free the stream.
                 // We will return Success to the MsQuic to prevent double free.
                 streamHandle.Dispose();
             }
 
-            return MsQuicStatusCodes.Success;
+            return QUIC_STATUS_SUCCESS;
```

Comment:
Idea for refactoring (separately from this change): pass ref to the associated data instead of the entire union.

```suggestion
       private static unsafe int HandleEventNewStream(State state, ref PEER_STREAM_STARTED event)
        {
            var streamHandle = new SafeMsQuicStreamHandle(event.Stream);
            if (!state.TryQueueNewStream(streamHandle, event.Flags))
            {
                // This will call StreamCloseDelegate and free the stream.
                // We will return Success to the MsQuic to prevent double free.
                streamHandle.Dispose();
            }

            return QUIC_STATUS_SUCCESS;
```

Reply:
For that, we'd need to get reasonable named structs for the anonymous types, now the type name is: `_PEER_STREAM_STARTED_e__Struct`.

Reply:
And changing the name of the anonymous types is basically a non starter, unless we can get ClangSharp to give us a way to change the format.

Reply:
I personally don't think the name is a dealbreaker for that, unless it can change later which would need to be fixed after a possible update.